From c3d36fdd2c4fbce32b98316d9851c5cf2db8efb4 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 19 Mar 2021 09:35:11 +0200
Subject: [PATCH 1/1] systemctl: Don't emit a reload warning for
 systemd-cryptsetup

This is modified by dracut during early boot, so instead of a nice clean prompt for the password, the user is presented with a scary warning about things being modified on disk, that they cannot possibly understand isn't actually a bad thing.

This simply skips all reload warnings for the transient dracut unit on Solus for systemd-cryptsetup, as this was generated *only* for the boot unit.
---
 src/systemctl/systemctl-start-unit.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/systemctl/systemctl-start-unit.c b/src/systemctl/systemctl-start-unit.c
index b398e77eb2..3aa5ba9703 100644
--- a/src/systemctl/systemctl-start-unit.c
+++ b/src/systemctl/systemctl-start-unit.c
@@ -143,8 +143,12 @@ static int start_unit_one(
                         return bus_log_parse_error(r);
         }
 
-        if (need_daemon_reload(bus, name) > 0)
+        /* Skip systemd-cryptsetup@ in early boot. No need for fear. */
+        if (need_daemon_reload(bus, name) > 0) {
+            if (!(strncmp(name, "systemd-cryptsetup@", 19) == 0 && arg_scope == UNIT_FILE_SYSTEM)) {
                 warn_unit_file_changed(name);
+            }
+        }
 
         if (w) {
                 log_debug("Adding %s to the set", path);
-- 
2.30.2

